src/js/JournalContainer.js

```diff
@@ -46,19 +66,42 @@ const JournalTextEntry = () => {
   };
 
   const saveEntry = () => {
-    let timestamp = Date.now();
+    let entries = [...journalEntries];
+    let timestamp = Date(Date.now());
 
     const journalEntry = {
+      journal_id: id == undefined ? uuidv4() : findJournalEntry(id).journal_id,
+      prompt: document.getElementsByClassName('prompt')[0].innerHTML,
       title: title,
       content: value,
-      date: timestamp
+      date: timestamp,
+      type: id == undefined ? journalType : findJournalEntry(id).type
+    };
+
+    const spliceIndex = () => {
+      entries.splice(entries.indexOf(id), 1);
+      entries.push(journalEntry);
     };
 
-    localStorage.setItem('journalEntry', JSON.stringify(journalEntry));
+    id == undefined ? entries.push(journalEntry) : spliceIndex();
+
+    localStorage.setItem('journalEntry', JSON.stringify(entries));
+  };
+
+  const displayPrompt = () => {
+    switch (journalType) {
+      case '/prompt-journal':
+        return <GeneratePrompt />;
+      case '/free-journal':
+        return <GenerateSubHeader />;
```

Comment:
The react dom refreshes anytime one types, which can be distracting. Maybe using useState can help prevent this.